src/shadowcasting.cpp

```diff
@@ -494,131 +494,131 @@ void cast_zlight(
-        output_caches, input_arrays, floor_caches, origin, offset_distance, numerator );
-    //   @
-    //  ..
-    // ...
-    cast_horizontal_zlight_segment < -1, 0, 0, 1, 1, T, calc, is_transparent, accumulate > (
-        output_caches, input_arrays, floor_caches, origin, offset_distance, numerator );
-    //   .
-    //  ..
-    // @..
-    cast_horizontal_zlight_segment < 0, 1, -1, 0, 1, T, calc, is_transparent, accumulate > (
-        output_caches, input_arrays, floor_caches, origin, offset_distance, numerator );
-    // ...
-    // ..
-    // @
-    cast_horizontal_zlight_segment < 1, 0, 0, -1, 1, T, calc, is_transparent, accumulate > (
-        output_caches, input_arrays, floor_caches, origin, offset_distance, numerator );
-    // .
-    // ..
-    // ..@
-    cast_horizontal_zlight_segment < 0, -1, -1, 0, 1, T, calc, is_transparent, accumulate > (
-        output_caches, input_arrays, floor_caches, origin, offset_distance, numerator );
-    // ...
-    //  ..
-    //   @
-    cast_horizontal_zlight_segment < -1, 0, 0, -1, 1, T, calc, is_transparent, accumulate > (
-        output_caches, input_arrays, floor_caches, origin, offset_distance, numerator );
-
-    // Straight up
-
-    // @.
-    // ..
-    cast_vertical_zlight_segment < 1, 1, 1, T, calc, is_transparent, accumulate > (
-        output_caches, input_arrays, floor_caches, origin, offset_distance, numerator );
-    // ..
-    // @.
-    cast_vertical_zlight_segment < 1, -1, 1, T, calc, is_transparent, accumulate > (
-        output_caches, input_arrays, floor_caches, origin, offset_distance, numerator );
-    // .@
-    // ..
-    cast_vertical_zlight_segment < -1, 1, 1, T, calc, is_transparent, accumulate > (
-        output_caches, input_arrays, floor_caches, origin, offset_distance, numerator );
-    // ..
-    // .@
-    cast_vertical_zlight_segment < -1, -1, 1, T, calc, is_transparent, accumulate > (
-        output_caches, input_arrays, floor_caches, origin, offset_distance, numerator );
-
-    // Straight down
-
-    // @.
-    // ..
-    cast_vertical_zlight_segment < 1, 1, -1, T, calc, is_transparent, accumulate > (
-        output_caches, input_arrays, floor_caches, origin, offset_distance, numerator );
-    // ..
-    // @.
-    cast_vertical_zlight_segment < 1, -1, -1, T, calc, is_transparent, accumulate > (
-        output_caches, input_arrays, floor_caches, origin, offset_distance, numerator );
-    // .@
-    // ..
-    cast_vertical_zlight_segment < -1, 1, -1, T, calc, is_transparent, accumulate > (
-        output_caches, input_arrays, floor_caches, origin, offset_distance, numerator );
-    // ..
-    // .@
-    cast_vertical_zlight_segment < -1, -1, -1, T, calc, is_transparent, accumulate > (
-        output_caches, input_arrays, floor_caches, origin, offset_distance, numerator );
+    if( static_cast<int>( dir ) | static_cast<int>( vertical_direction::DOWN ) ) {
```

Comment:
IMO this is much more readable.
```suggestion
    if( dir == vertical_direction::DOWN || dir == vertical_direction::BOTH ) {
```